client/ayon_maya/api/lib.py

```diff
@@ -4377,6 +4386,66 @@ def create_rig_animation_instance(
         )
 
 
+def create_camera_instance(
+        nodes, context, namespace, options=None, log=None
+):
+    """Create a camera instance from the loaded camera nodes.
+
+    Args:
+        nodes (list): The loaded camera nodes.
+        context (dict): Representation context of the camera container.
+        namespace (str): The namespace for the new camera instance.
+        options (dict, optional): Additional options for the camera instance.
+        log (logging.Logger, optional): Logger to log to if provided.
+    """
+    if options is None:
+        options = {}
+
+    camera_nodes = [
+        node for node in nodes if cmds.referenceQuery(
+        node, isNodeReferenced=True) and
+        cmds.listRelatives(node, children=True, type="camera")
+    ]
```

Comment:
This will be relatively slow because you're doing `listRelatives` call for each node individually.

Also, `listRelatives` suffers from the fact that without `fullPath=True` it by defaults does not return unique paths if there are objects with same names in the scene.

And note that `listRelatives` returns `None` if there are no valid things to return... (odd API in maya I know). We are only passing it to `cmds.select` which takes `None` just fine, regardless I'd prefer the type to be consistent so use `or []`

So I'd do:
```suggestion
    referenced_nodes: list[str] = [
        node for node in nodes 
        if cmds.referenceQuery(node, isNodeReferenced=True)
    ]
    camera_nodes: list[str] = cmds.listRelatives(
        referenced_nodes, children=True, type="camera", fullPath=True
    ) or []
```